### PR TITLE
Steam 863/planned treatment phase viz

### PR DESCRIPTION
**Pull request by Dtphelan1**
Adds mapping and visualization of PlannedTreatmentPhase profiled ServiceRequest resources based on the IG [here](https://build.fhir.org/ig/HL7/codex-radiation-therapy/StructureDefinition-codexrt-radiotherapy-planned-course.html).

A few other small changes: 
- Change the name of `mapPhase` to `mapTreatedPhase`, more closely aligning with the profile name;
- Move all profile-specific visualizers into a nested directory for better organization of files; I was getting a bit overwhelmed scrolling through a list of 15+ files to find the correct one in the DataView file, and this segments the generic viewer components from the business-logic specific profileVisualizers. 
- Move the iterate-over-mapped-data logic from the DataView to the TreatmentPhaseTable component; the DataView component shouldn't need to to know anything about how the treatmentPhaseTable data is organized.

### CODE REVIEW

File: src/lib/mappingUtils.js [hunk 1]
```diff
@@ -99,10 +99,63 @@ function mapPhase(procedure) {
...
+    // IG states there will be at most one procedure intent
+    output["Modalities"] = fhirpath.evaluate(
+      plannedPhase,
+      "ServiceRequest.extension.where(url = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-radiotherapy-modality-and-technique').extension.where(url = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-radiotherapy-modality').valueCodeableConcept.coding.display"
+    )[0];
+    const technique = fhirpath.evaluate(
+      plannedPhase,
+      "ServiceRequest.extension.where(url = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-radiotherapy-modality-and-technique').extension.where(url = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-radiotherapy-technique').valueCodeableConcept.coding"
+    )[0];
+    output["Techniques"] = technique ? technique.display : undefined;
```
Comment (julianxcarter) [inline src/lib/mappingUtils.js]: I think it'd be best to alter the above fhirpath to return the `technique.display` value directly, just to be sure that nothing breaks down if the display isn't populated. Similar to what you're doing with the `Modalities` fetching.

**Review by julianxcarter — CHANGES_REQUESTED** [review]
This looks good and works well! I had two comments, one of which I couldn't actually write as part of the review:

Do we think we should remove `Patient-XRTS-01` and `Patient-XRTS-02` as default queries? They seem redundant now that we have the more coplete examples.

File: src/lib/mappingUtils.js (hunk 1 above)
Comment (Dtphelan1) [reply src/lib/mappingUtils.js]: Completely agree. This is what happens when someone blindly copy and pastes from previous examples. I'll correct that here and from the source

**Comment by Dtphelan1** [comment]
> Do we think we should remove Patient-XRTS-01 and Patient-XRTS-02 as default queries? They seem redundant now that we have the more coplete examples.

I think actually once we add the missing-data sections, they will be useful patients to illustrate what gaps look like. I do agree that they're not the gold-source patients that we used to think of them as. Thoughts on relegating them to the bottom of the ID list, so they're not the first thing you have to see?

**Comment by julianxcarter** [comment]
> I think actually once we add the missing-data sections, they will be useful patients to illustrate what gaps look like. I do agree that they're not the gold-source patients that we used to think of them as. Thoughts on relegating them to the bottom of the ID list, so they're not the first thing you have to see?

That's a great point and I hadn't thought of that! I do think I'd prefer them being at the bottom of the list.

**Review by julianxcarter — APPROVED** [review]

**Pull request merged**